nncf/torch/quantization/quantize_model.py

```diff
@@ -108,7 +109,8 @@ def quantize_impl(
     if target_device == TargetDevice.CPU_SPR:
         raise RuntimeError("target_device == CPU_SPR is not supported")
 
-    nncf_network = create_nncf_network(model.eval(), calibration_dataset)
+    copied_model = deepcopy(model)
+    nncf_network = create_nncf_network(copied_model.eval(), calibration_dataset)
```

Comment:
Could you please explain the reason behind this change?

Reply:
@AlexanderDokuchaev, could you please reply?

Reply:
All implementation of `quantize` function works with copied of the model. 
https://github.com/openvinotoolkit/nncf/blob/release_v250/nncf/torch/quantization/quantize_model.py#L252 previous implementation for torch backend.
For new `quantize_impl` it was missed.